### ISSUE TITLE
splash page for hackweek web template

### ISSUE THREAD

**Issue by aaarendt** [uwhackweek/jupyterbook-template#38]
We aim to have a static html "splash" page with event details, organizers and sponsors.

**Comment by aaarendt** [comment]
After merging in Don's work (#39) he suggested the pathway for this is to deploy the template to `some-hackweek` organization and then use actions to deploy the website to `some-hackweek.github.io`. These actions would perform the building of the Jupyterbook and the deployment of cookie cutter (and associated file management). Separate branches could be used for each hackweek instance and the `index.html` could also be relabeled to match the year of the hackweek (e.g. `2022.html`).

**Comment by aaarendt** [comment]
Some clarifications:

* the template would be imported each time a new event was hosted, so that it includes all the latest updates from the main UW hackweek page
* the template would have actions to deploy the html on the separate `some-hackweek.github.io` (see [here](https://cookiecutter.readthedocs.io/en/1.7.2/usage.html#works-directly-with-git-and-hg-mercurial-repos-too))
* there would be some system established on `some-hackweek.github.io` to manage/archive different hackweek instances using branches 
* so the template would just act as a temporary generator of the html containing the web content, which is stored persistently in the `some-hackweek.github.io` repo. The code used to generate the web content each year would not persist, but this should be fine?

**Comment by JessicaS11** [comment]
Thanks for these clarifications, @aaarendt! 

It sounds like it we could then - in the actions to generate/update the website content (which ultimately lives in a separate repo from the JupyterBook) - require tags or branches with older content. So next steps on this are:
- [ ] merge #39 
- [ ] write actions to run cookiecutter for testing/previewing the site
- [ ] run cookiecutter and publish the site
- [ ] check for existing tags/branches before updating site content

**Comment by aaarendt** [comment]
Thanks for creating this list @JessicaS11! I have one task to add, which is to include a small Python application that converts the cookiecutter configuration file from yaml to json (I converted Don's existing json file to yaml on his recommendation and it's much easier to work with, plus we can comment it for new hackweek creators).

Would you like me to add this to #39 , or do this later in a separate PR?

**Comment by jomey** [comment]
I think all the remaining ToDo's have been addressed?

**Comment by JessicaS11** [comment]
> I think all the remaining ToDo's have been addressed?

I think technically not yet:

> then we could require tags/branches to mark older content on the organization.github.io page, which lives in an organization.github.io repo that is updated by the organization/jupyterbook repo (which is in turn templated per-event from the most recent uwhackweek/jupyerbook-template repo (this repo)).

We haven't created actions that do this so that we can tag/store versions of the site in cases where there are multiple events. But, it's slightly outside the spirit of this issue title at this point, so I'll move it to a new issue. We can work on it within the IS2 event repo, which is where it will first be needed (but not until after the event).
